vispy/visuals/box.py

```diff
@@ -43,28 +43,43 @@ class BoxVisual(MeshVisual):
     def __init__(self, width=1, height=1, depth=1, width_segments=1,
                  height_segments=1, depth_segments=1, planes=None,
                  vertex_colors=None, face_colors=None,
-                 color=(0.5, 0.5, 1, 1), edge_color=None):
+                 color=(0.5, 0.5, 1, 1), edge_color=None, **kwargs):
         vertices, filled_indices, outline_indices = create_box(
             width, height, depth, width_segments, height_segments,
             depth_segments, planes)
 
-        MeshVisual.__init__(self, vertices['position'], filled_indices,
-                            vertex_colors, face_colors, color)
+        self._mesh = MeshVisual(vertices['position'], filled_indices,
+                                vertex_colors, face_colors, color)
         if edge_color:
-            self._outline = MeshVisual(vertices['position'], outline_indices,
-                                       color=edge_color, mode='lines')
+            self._border = MeshVisual(vertices['position'], outline_indices,
+                                      color=edge_color, mode='lines')
         else:
-            self._outline = None
+            self._border = MeshVisual()
 
-    def draw(self, transforms):
-        """Draw the visual
+        self._border.gl_state = {
+            "polygon_offset_fill": True,
```

Comment:
Sorry, I am referring to `Visual.update_gl_state()` or `Visual.set_gl_state()`.

* `border.gl_state` is not an attribute that is recognized by Visual; this will just be ignored.
* Read over `Visual.draw()`, understand how it works. Each visual has its own GL state variables, and these are applied immediately before each visual is drawn.